Tasks/Common/azure-arm-rest/azure-arm-storage.ts

```diff
@@ -122,6 +122,27 @@ export class StorageAccounts {
         return deferred.promise;
     }
 
+    public async _getStorageAccountDetails(storageAccountName: string, credentials: msRestAzure.ApplicationTokenCredentials, subscriptionId: string): Promise<StorageAccountInfo> {
```

Comment:
this doesn't look like the right method for this file which is basically a client layer. There are other issues with the signature: It shouldn't start with _. There should be no parameter like creds, or subcriptionId, StorageAccountInfo doesn't seem like the generic object

Reply:
instead have a method get(storageAccountName: string): Promise<Model.StorageAccount>.
In the caller, when you need the key, use two calls there.